include/OpenSpaceToolkit/Physics/Environment.hpp

```diff
@@ -161,16 +190,34 @@ class Environment
     ///                     Contains Earth, Sun and Moon, with SPICE-based ephemeris.
     ///
     /// @code
-    ///                     Environment environment = Environment::Default() ;
+    ///                     Environment environment = Environment::Default();
     /// @endcode
+    /// @param              [in] setGlobalInstance True if the global environment instance should be set with the
+    /// default
     ///
     /// @return             Undefined environment
 
-    static Environment Default();
+    static Environment Default(const bool& setGlobalInstance = false);
+
+    /// @brief              Get the singleton instance of the environment
+    ///
+    /// @return             Shared pointer to the singleton instance
+
+    static Shared<Environment> AccessInstance();
+
+    /// @brief              Set the singleton instance of the environment
+    ///
+    /// @param              [in] instance Shared pointer to the new singleton instance
+
+    static void SetInstance(const Shared<Environment>& instance);
 
    private:
     Instant instant_;
     Array<Shared<const Object>> objects_;
+    Shared<const Object> centralCelestial_;
+
+    static std::shared_mutex mutex_;
+    static Shared<Environment> instance_;
```

Comment:
Should these static properties should be part of the object itself? I think they should better live outside no?